### PR TITLE
Age Restriction Workaround

### LINKED ISSUE #52: Can't download a specific video (id=whoI4EuC0yg)
### Summary

Error downloading whoI4EuC0yg: YouTube returned an unexpected response: `Could not acquire the player response from the watch html!
It looks like YouTube changed it's API again :-/
If this not yet reported, it would be great if you could file an issue:
            (https://github.com/DzenanJupic/rustube/issues/new?assignees=&labels=youtube-api-changed&template=youtube_api_changed.yml).`

### Video IDs

- whoI4EuC0yg

### Version

```text
- 0.6.0 (https://github.com/Discursif/rustube fork to allow downloading at high-speed)
```

### Log output or error

[log.txt](https://github.com/DzenanJupic/rustube/files/10247613/log.txt)

### Additional notes or ideas

Maybe it's due to the high speed downloading fix I applied but this is the only video unable to get downloaded

### PR DESCRIPTION
**Pull request by mjmjelde**
A potential age restriction workaround.  First Rust contribution I've made, so please go easy on my attempt at a workaround.

Potentially fixes #52 and #38

### CODE REVIEW

**Comment by DzenanJupic** [comment]
Hey @mjmjelde, thanks for the PR and welcome to the Rust community!

I'm somewhat sorry for you that your first contribution has to be a codebase like this. I wrote it quite some time ago, and it's not especially clean 😅.

I'll test your PR and get back to you soon.

**Comment by mjmjelde** [comment]
Thanks for the welcome 😄 

And don't be sorry, been able to learn quite a bit from your code.  Hasn't been hard to follow what is happening at all.

Just pushed some clippy error fixes for what failed in the checks
